### PR TITLE
Changed DataTable to add `fill` and `pin`

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed DataTable to add `fill` and `pin`.
`fill` allows the table to dynamically respond to the available real estate.
`pin` at the top level allows the header and/or footer to be pinned.
`pin` is also available in `columns` to pin the first column.

#### Where should the reviewer start?

index.d.ts

#### What testing has been done on this PR?

Added Fill story. Tested across browsers and at different screen resolutions.

#### How should this be manually tested?

Fill story

#### Do the grommet docs need to be updated?

automatic

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible - even though there are some snapshot changes, they don't appear to be visible

### CODE REVIEW

File: src/js/components/DataTable/index.d.ts [hunk 1]
```diff
@@ -58,9 +60,11 @@ export interface DataTableProps<TRowType = any> {
     | Sections<BackgroundType | string[], BackgroundType, BackgroundType>;
   border?: BorderType | Sections<BorderType>;
   columns?: ColumnConfig<TRowType>[];
+  fill?: boolean | 'vertical' | 'horizontal';
   gridArea?: GridAreaType;
   margin?: MarginType;
   pad?: PadType | Sections<PadType>;
+  pin?: boolean | 'header' | 'footer';
```
Comment (ShimiSun) [inline src/js/components/DataTable/index.d.ts:67]: Duplication of line 43?

File: src/js/components/Table/StyledTable.js [hunk 2]
```diff
@@ -60,9 +59,7 @@ const StyledTableDataCaption = styled.caption`
 StyledTableDataCaption.defaultProps = {};
 Object.setPrototypeOf(StyledTableDataCaption.defaultProps, defaultProps);
 
-const StyledTableRow = styled.tr`
-  height: 100%;
-`;
+const StyledTableRow = styled.tr``;
```
Comment (ShimiSun) [inline src/js/components/Table/StyledTable.js:62]: Was the height safe for removal?

File: src/js/components/DataTable/README.md [hunk 3]
```diff
@@ -396,6 +413,18 @@ Use this to indicate that 'data' doesn't contain all that it could.
 function
 ```
 
+**pin**
+
+Whether the header and/or footer should be pinned when
+      not all rows are visible. This only takes effect when 'fill' is true
```
Comment (ShimiSun) [inline src/js/components/DataTable/README.md:419]: I could not find where is enforcement of the last sentence, I'll take another look in a few hours.

File: src/js/components/DataTable/index.d.ts (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/index.d.ts:67]: Nope. Line 43 is for pinning a column for horizontal scrolling. Line 67 is pinning the header and/or footer for vertical scrolling.

File: src/js/components/Table/StyledTable.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/Table/StyledTable.js:62]: I believe so. I've run through a bunch of scenarios to validate and haven't uncovered any issues yet.

File: src/js/components/DataTable/README.md (hunk 3 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/README.md]: removed

**Comment by ShimiSun** [comment]
Linking to https://github.com/grommet/grommet/issues/4431

File: src/js/components/DataTable/doc.js [hunk 4]
```diff
@@ -163,6 +169,13 @@ export const doc = DataTable => {
       browser, such as columns.search, sortable, groupBy, or 
       columns.aggregate.`,
     ),
+    pin: PropTypes.oneOfType([
+      PropTypes.bool,
+      PropTypes.oneOf(['header', 'footer']),
+    ]).description(
+      `Whether the header and/or footer should be pinned when
+      not all rows are visible.`,
+    ),
```
Comment (ShimiSun) [inline src/js/components/DataTable/doc.js:178]: It is not clear what is the default behavior when `pin=true`.
Also, how do you set both to be pinned if the proptypes are limiting to a string or a bool?

File: src/js/components/DataTable/doc.js [hunk 5]
```diff
@@ -23,7 +23,7 @@ parts.forEach(part => {
 });
 
 const backgroundShape = {};
-parts.forEach(part => {
+[...parts, 'pinned'].forEach(part => {
```
Comment (ShimiSun) [inline src/js/components/DataTable/doc.js:26]: There is no description of what it means to 'pin' a single column. I'm not sure I understand what it means.
Edit: I just read one of your previous review comments and I understand that pin for a column is for pinning a column for horizontal scrolling. I think we should mention that.

File: src/js/components/DataTable/stories/Fill.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+          header: { color: 'background-front', opacity: 'strong' },
+          footer: { color: 'background-front', opacity: 'strong' },
+          pinned: { color: 'background-front', opacity: 'strong' },
+        }}
+      />
+    </Box>
+  </Grommet>
+);
+
+storiesOf('DataTable', module).add('Fill', () => <Example />);
```
Comment (ShimiSun) [inline src/js/components/DataTable/stories/Fill.js:33]:
```suggestion
storiesOf('DataTable', module).add('Fill & pin', () => <Example />);
```

File: src/js/components/DataTable/doc.js (hunk 4 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/doc.js:178]: added more detail

File: src/js/components/DataTable/doc.js (hunk 5 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/doc.js:26]: added documentation

**Pull request merged**